### PR TITLE
436385 accept licence

### PR DESCRIPTION
**Pull request by jbarnsley10**

### CODE REVIEW

File: app/routes/user.js [hunk 1]
```diff
@@ -21,6 +22,46 @@ module.exports = [
...
+
+      return h.response(res).code(200)
+    }
+  },
+  {
+    method: 'PUT',
+    path: '/user/me/licence',
+    options: {
+      tags: ['api'],
+      notes: ['Checks if the calling user has accepted the licence'],
```
Comment (whitewaterdesign) [inline app/routes/user.js:50]: notes needs updating

File: app/routes/user.js [hunk 2]
```diff
@@ -21,6 +22,46 @@ module.exports = [
...
+      response: {
+        status: {
+          200: undefined,
+          401: undefined,
+          404: undefined
+        }
```
Comment (whitewaterdesign) [inline app/routes/user.js:56]: does the sequelize query handle the 404 itself?  we probably don't need 401 here.

File: app/service/registration.js [hunk 3]
```diff
@@ -69,10 +69,9 @@ class RegistrationService {
       if (account.activation_token_expiry <= new Date()) {
         return actionResults.ACTIVATION_CODE_EXPIRED
       } else {
-        account.activated_date = new Date() // UTC ??
-        await account.save()
+        await this.userAccountRepository.setActivatedDate(username)
```
Comment (whitewaterdesign) [inline app/service/registration.js:72]: else is not needed here

File: app/routes/user.js (hunk 2 above)
Comment (jbarnsley10) [reply app/routes/user.js:56]: Removed 401

File: app/routes/user.js (hunk 1 above)
Comment (jbarnsley10) [reply app/routes/user.js:50]: Updated note

File: app/routes/user.js (hunk 2 above)
Comment (jbarnsley10) [reply app/routes/user.js:56]: CHanged error condition to 500

File: test/unit/app/dto/licence.test.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  })
+
+  const request = {
+    auth: {
+      artifacts: {
+        decoded: {
+          header: { alg: 'RS256', typ: 'JWT', kid: 'aphw-ddi-enforcement' },
+          payload: {
+            scopes: ['Dog.Index.Enforcement'],
+            username: 'dev-user@test.com',
```
Comment (whitewaterdesign) [inline test/unit/app/dto/licence.test.js:20]: this will become scope after the PR goes through

**Review by whitewaterdesign — APPROVED** [review]
👍

File: app/service/registration.js [hunk 5]
```diff
@@ -98,14 +97,31 @@ class RegistrationService {
...
+
+    if (!account) {
+      return actionResults.ACCOUNT_NOT_FOUND
+    }
+
+    if (!account.active) {
+      return actionResults.ACCOUNT_NOT_ENABLED
+    }
+
+    return await this.userAccountRepository.setLicenceAcceptedDate(username) ? actionResults.OK : actionResults.ERROR
```
[the remark this answers is not in the document]
Comment (whitewaterdesign) [reply app/service/registration.js:124]: my bad, didn't see the rest of the line

**Review by whitewaterdesign — CHANGES_REQUESTED** [review]
just that scopes needs changing to scope

**Review by whitewaterdesign — APPROVED** [review]
👍

**Pull request merged**